firebase.go

```diff
@@ -105,6 +105,22 @@ func (fb *Firebase) Unauth() {
 	fb.params.Del(authParam)
 }
 
+// Ref returns a copy of an existing Firebase reference with a new path.
+func (fb *Firebase) Ref(path string) *Firebase {
+	newFB := fb.copy()
+	parsedURL, err := _url.Parse(fb.url)
+	if err != nil {
+		panic(err)
```

Comment:
My only concern here is this panic. Could we return an error if we failed to parse the url instead? After that, I think this is good to go.